src/main/java/org/genomicsdb/spark/GenomicsDBSchemaFactory.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * The MIT License (MIT)
+ * Copyright (c) 2020 Omics Data Automation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * */
+
+package org.genomicsdb.spark;
+
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.sql.types.*;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import scala.collection.JavaConverters;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Collection;
+import java.util.Set;
+import java.util.HashSet;
+
+/**
+ * Defines the default schema and provides methods to extend upon the default schema. 
+ * Can build a vid map that is used for GenomicsDBInput, and will also ensure that types 
+ * are validated using DataType DDL.
+ *
+ * When a vid map is needed, an instance should be defined with a constructor. When 
+ * the vid map is not needed, such as from the scala api - then this is treated as a 
+ * utility class.
+ **/
```

Comment:
I'm guessing this is something to do with DataFrame and that this particular 'schema' information is not needed by other interfaces (Python etc)

Reply:
The schema information is needed for the DataSource regardless of client language type. Currently python and R use default schema, but in the future python will have a schema factory as well that supports extended the schema from the client like this.

Reply:
I see - is there any way to not duplicate the factory functionality across languages (apart from a simple wrapper class)?